### ISSUE TITLE
Complementary conditions in OR get evaluated without checking whether they're further ANDed with something else

### ISSUE THREAD

**Issue by lserni** [kalessil/phpinspectionsea#1849]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
Plugin: EA ultimate 2021.5
Language level: PHP 7.4

Suppose I have an invoice which may or may not include VAT; in the latter case I need exactly two invoice lines, in the former I need at least one but may have more. The check I have to do is:

    if the invoce has VAT,
         and has less than 1 line,
    or it has no VAT,
         and hasn't exactly two lines,
    THEN
         this invoice is wrong.

This becomes:

```
   $VAT   = (bool)checkVat();
   $lines = (array)getLines();

   if (($VAT === true) && (count($lines) < 1))
       || (($VAT !== true) && (count($lines) !== 2))
   ) {
       doSomething();
   }

```

Current behaviour:

An issue is raised:

       [EA] '$VAT === true || $VAT !== true' seems to be always true. 

Expected behaviour

      No issues are raised

Note: the above is a contrived example and the issue may be avoided by replacing "$VAT !== true" with "$VAT === false" (which, $VAT being boolean, is in itself strange). The real example can't be worked around so simply, the best I've come up with to avoid warnings is,

    $conditionSatisfied = 0;
    if (...) { $conditionSatisfied++; }
    if (...) { $conditionSatisfied++; }
    if ($condition) {
        doSomething();
    }
